Deprecating two-branch-handling, release 1.3.0

eschricker opened the pull request:
Develop will be merged into main. Afterwards main will be set as default branch and the develop-branch will be deleted. The new state will be tagged as 1.3.0.

mfn commented on src/Claims/Claim.php:
```diff
@@ -146,6 +148,7 @@ public function matches($value, $strict = true)
      *
      * @return array
      */
+    #[\ReturnTypeWillChange]
```
changelog candidate

mfn commented on src/Http/Middleware/BaseMiddleware.php:
```diff
@@ -24,16 +25,12 @@ abstract class BaseMiddleware
-     * @var JWTAuth
      */
-    protected $auth;
+    protected JWTAuth $auth;
```
Shouldn't be a problem, or? What if someone overrode this Middleware and also defined this property? 🤔

---
Note: _Guess it's fine_, also this isn't the only property being typed now in this PR but probably also something for the changelog.

mfn commented on src/JWTGuard.php:
```diff
@@ -503,17 +483,20 @@ protected function fireAttemptEvent(array $credentials)
-            $this->name,
-            $user
-        ));
+        if (class_exists('Illuminate\Auth\Events\Validated')) {
```
note for changelog

mfn reviewed (approved):
LGTM!

Went through all the non-test source file changes diligently; since everything was already approved in PRs before we're fine.

Before releasing 1.3.0 we should merge https://github.com/PHP-Open-Source-Saver/jwt-auth/pull/93 and generally update the changelog (seems we/someone has to do this manual; I can do it later today/night if no one beats me to it).

leon0399 commented:
> we should merge #93 and generally update the changelog

I've added missing changelog entries stated above in #93, if you can find more - feel fre to add them! 😃

eschricker commented:
I have merged #93. Now we can merge this PR and create the new version or is something still missing? :)

leon0399 commented on CHANGELOG.md:
```diff
@@ -0,0 +1,44 @@
+
+You can find and compare releases at the GitHub release page.
+
+## [Unreleased]
```
Guess it's needed to be transfered to [1.3.0] on merge

mfn commented:
@eschricker 😄 

![image](https://user-images.githubusercontent.com/87493/149332732-526e8773-d7d7-49a5-abf3-6ecf7d25301c.png)

eschricker commented:
I have updated the changelog on main and added the tag for 1.3.0.

The default branch has been changed to main. The two open PRs have been changed to target main instead of develop. 

Thank you all for your support :)

mfn commented:
@eschricker thank you for doing all the work 🙏🏼
